File: backend/src/main/java/dev/saul/descarbonizacion_transporte/modules/equipments/dto/output/EquipmentOutputDto.java
package dev.saul.descarbonizacion_transporte.modules.equipments.dto.output;

import java.util.List;

import dev.saul.descarbonizacion_transporte.modules.equipments.entity.EquipmentDataField;
import dev.saul.descarbonizacion_transporte.modules.type_equipments.entity.DataField;
import dev.saul.descarbonizacion_transporte.modules.type_equipments.entity.TypeEquipmentEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EquipmentOutputDto {
    private String id;
    private String name;
    private String description;
    private List<EquipmentDataField> data;
    private List<DataField> otherDataFields;
    private TypeEquipmentEntity typeEquipment;
    private String usageDataEnergyConsumedMeasurementUnit;
    private String usageDataLoadOrCapacityUsedMeasurementUnit;
}

File: frontend/src/api/equipments/interfaces/output/equipment.output.dto.ts
import type EquipmentDataFieldInputDto from '@/api/equipments/interfaces/input/equipment-data-field.input.dto.ts'
import type DataFieldInputDto from '@/api/typeEquipments/interfaces/input/data-field.input.dto.ts'
import type TypeEquipmentOutputDto from '@/api/typeEquipments/interfaces/output/type-equipment.output.dto.ts'

export default interface EquipmentOutputDto {
  id: string
  name: string
  description: string
  data: EquipmentDataFieldInputDto[]
  otherDataFields: DataFieldInputDto[]
  typeEquipment: TypeEquipmentOutputDto
  usageDataEnergyConsumedMeasurementUnit: string
  usageDataLoadOrCapacityUsedMeasurementUnit: string
}
